src/formats/png.zig

```diff
@@ -1,476 +1,45 @@
-        } else {
-            return 0;
-        }
-    }
-
-    inline fn getB(self: Self, index: usize, current_row: []const u8, previous_row: []const u8) u8 {
-        _ = self;
-        _ = current_row;
-        return previous_row[index];
-    }
-
-    inline fn getC(self: Self, index: usize, current_row: []const u8, previous_row: []const u8) u8 {
-        _ = current_row;
-        if (index >= self.pixel_stride) {
-            return previous_row[index - self.pixel_stride];
-        } else {
-            return 0;
-        }
-    }
-
-    fn paethPredictor(a: u8, b: u8, c: u8) ImageReadError!u8 {
-        const large_a = @intCast(isize, a);
-        const large_b = @intCast(isize, b);
-        const large_c = @intCast(isize, c);
-        const p = large_a + large_b - large_c;
-        var pa = p - large_a;
-        var pb = p - large_b;
-        var pc = p - large_c;
-        if (pa < 0) pa = -pa;
-        if (pb < 0) pb = -pb;
-        if (pc < 0) pc = -pc;
+const Allocator = std.mem.Allocator;
 
-        if (pa <= pb and pa <= pc) {
-            return @intCast(u8, large_a & 0xFF);
-        } else if (pb <= pc) {
-            return @intCast(u8, large_b & 0xFF);
-        } else {
-            return @intCast(u8, large_c & 0xFF);
-        }
-    }
-};
+pub const HeaderData = types.HeaderData;
+pub const ColorType = types.ColorType;
+pub const CompressionMethod = types.CompressionMethod;
+pub const FilterMethod = types.FilterMethod;
+pub const FilterType = types.FilterType;
+pub const InterlaceMethod = types.InterlaceMethod;
+pub const any_chunk_type = reader.any_chunk_type;
+pub const any_chunk_type_id = reader.any_chunk_type_id;
+pub const isChunkCritical = reader.isChunkCritical;
+pub const load = reader.load;
+pub const loadHeader = reader.loadHeader;
+pub const loadWithHeader = reader.loadWithHeader;
+pub const ChunkProcessData = reader.ChunkProcessData;
+pub const PaletteProcessData = reader.PaletteProcessData;
+pub const RowProcessData = reader.RowProcessData;
+pub const ReaderProcessor = reader.ReaderProcessor;
+pub const TrnsProcessor = reader.TrnsProcessor;
+pub const PlteProcessor = reader.PlteProcessor;
+pub const ReaderOptions = reader.ReaderOptions;
+pub const DefProcessors = reader.DefProcessors;
+pub const DefOptions = reader.DefOptions;
+pub const required_temp_bytes = reader.required_temp_bytes;
 
-// Remember, PNG uses network byte order (aka Big Endian)
-// TODO: Proper validation of chunk order and count
 pub const PNG = struct {
-    header: IHDR = undefined,
-    chunks: std.ArrayList(ChunkVariant) = undefined,
-    allocator: Allocator = undefined,
-
-    const DecompressionContext = struct {
-        pixels: *color.PixelStorage = undefined,
-        pixels_index: usize = 0,
-        compressed_data: std.ArrayList(u8) = undefined,
-        filter: PngFilter = undefined,
-        x: usize = 0,
-        y: usize = 0,
-    };
-
     const Self = @This();
 
     pub fn init(allocator: Allocator) Self {
```

Comment:
Remove the `allocator` parameter if you don't use it.

Reply:
This whole method is actually a left over from some experiment. I'll remove it.